GAE/src/com/gallatinsystems/framework/dao/BaseDAO.java

```diff
@@ -436,17 +431,34 @@ protected T findByProperty(String propertyName, Object propertyValue,
      * @return empty list if ids is null, otherwise a list of objects
      */
     public List<T> listByKeys(List<Long> ids) {
-        if (ids == null) {
+        return listByKeys(ids, concreteClass);
+    }
+
+    /*
+     * Retrieve a list of datastore entities by the keys provided
+     */
+    public List<T> listByKeys(List<Long> idsList, Class<T> clazz){
+        if (idsList == null || idsList.isEmpty()) {
             return Collections.emptyList();
         }
-        final List<T> list = new ArrayList<T>();
-        for (Long id : ids) {
-            final T obj = getByKey(id);
-            if (obj != null) {
-                list.add(obj);
-            }
+
+        PersistenceManager pm = PersistenceFilter.getManager();
+        List<Object> datastoreKeysList = new ArrayList<>();
+        for (Long id : idsList) {
+            Key key = KeyFactory.createKey(clazz.getSimpleName(), id);
+            Object objectId = pm.newObjectIdInstance(clazz, key);
+            datastoreKeysList.add(objectId);
         }
-        return list;
+
+        final List<T> resultsList = new ArrayList<>();
+
+        try {
+            resultsList.addAll(pm.getObjectsById(datastoreKeysList));
```

Comment:
What would happen if some of the ids are found and others are not, would it still work and returns the ones that were actually found or throw an exception?

Reply:
So I made a fallback method incase there is any wrong ID, the fall back retrieves the items one by one